File: Compiler/src/compiler/values/ComparisonResultValue.java
package compiler.values;

import compiler.Value;
import util.ComparisonResult;
import util.CompilationResult;
import util.DataType;
import util.JenaUtil;

public class ComparisonResultValue extends Value {
    public ComparisonResultValue(ComparisonResult value) {
        super(ComparisonResult.toString(value));
    }

    @Override
    public DataType resultDataType() {
        return DataType.COMPARISON_RESULT;
    }

    @Override
    public CompilationResult compile() {
        return new CompilationResult(JenaUtil.genStingVal(value), "", "");
    }
}

File: Compiler/src/compiler/values/ComparisonResultValue.kt
package compiler.values

import compiler.Value
import util.ComparisonResult
import util.CompilationResult
import util.DataType
import util.JenaUtil

class ComparisonResultValue(value: ComparisonResult) : Value(value.toString()) {

    override fun resultDataType(): DataType = DataType.ComparisonResult

    override fun compile(): CompilationResult =
        CompilationResult(JenaUtil.genLink(JenaUtil.POAS_PREF, value), emptyList(), "")
}